test/unit/http.js

```diff
@@ -0,0 +1,22 @@
+const sinon = require('sinon');
+
+const Http = require('../../src/http');
+
+describe('HTTP class', () => {
+  it('should allow me to provide my own fetch interface', async () => {
```

Comment:
Can we add a test to validate that the default `fetch` can return a different response than the custom `fetch` interface?

Reply:
+1 - we're testing that if we pass fetch, it will be called (and called with the expected args). 

What we are not testing (and should add coverage for):
- What happens if we don't pass fetch?
- What happens with the response from fetch? Our implementation is that requests will return the response, but it's not tested here.

In general, where are the rest of the tests for this class?

Reply:
The rest of this class is tested by proxy through our integration tests.  I agree that we need more unit tests for this class